programs/ssl/ssl_client2.c

```diff
@@ -2352,12 +2598,20 @@ int main( int argc, char *argv[] )
 #endif
 
 #if defined(MBEDTLS_ECP_C)
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+    if( named_groups_list[0] != MBEDTLS_ECP_DP_NONE )
+        mbedtls_ssl_conf_curves(&conf, named_groups_list);
+
+    if( key_share_named_groups_list[0] != MBEDTLS_ECP_DP_NONE )
+        mbedtls_ssl_conf_key_share_curves( &conf, key_share_named_groups_list );
+#else 
     if( opt.curves != NULL &&
```

Comment:
Same here: If I understand correctly, there should be a runtime check deciding which configuration API to call, not a compile-time check.